src/autocomplete.js

```diff
@@ -466,8 +466,10 @@ export default class Autocomplete extends Component {
         )}
 
         <input
+          aria-expanded={menuOpen ? 'true' : 'false'}
           aria-activedescendant={optionFocused ? `${id}__option--${focused}` : false}
           aria-owns={`${id}__listbox`}
+          aria-autocomplete='both'
```

Comment:
This change isn't mentioned in the commit message – does it have any perceivable effect? It seems like having moved the combobox role onto the text input we're following the [ARIA 1.0 pattern](https://www.w3.org/TR/wai-aria-1.0/roles#combobox) rather than [ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/#combobox). In ARIA 1.1, authors MUST add `aria-autocomplete`, but this is a SHOULD in the 1.0 pattern. If this doesn't provide any tangible benefit, I'd consider leaving this out.

If we're going to keep it, from reading [the spec](https://www.w3.org/TR/wai-aria-1.0/states_and_properties#aria-autocomplete), it sounds like this should only be `both` if `autoselect` is `true` – that's the only time when the autocomplete 'presents a value completion prediction inside the text input'. Otherwise, it should be `list`.

Reply:
Good point, I'd missed this (consideration of the config option of `both` vs `list`).
I can either remove, or introduce the dynamic value based upon config.

I added the extra attribute during JAWS testing at one point. It didn't improve the situation I was focused on, but I didn't subsequently remove it on the basis of added semantics.
As far as I'm aware it isn't having any direct impact upon any assistive tech behaviour currently, but I'll double check this on my next round of testing (once I've actioned these comments)

Regarding the combobox role and ARIA 1.0 vs 1.1 - in this case it seemed necessary to strike a balance between latest spec, and what actually works with the assistive tech. I suspect the spec is ahead of what the tech is currently supporting. Moving the combobox role onto the input and moving away from the 'composite' approach was the change that brought the field to the attention of the form navigation tools (where it was ignored previously).  It's an awkward one for sure - spec compliance vs real world compatibility with screen readers.
Happy to discuss further.

Reply:
Confirmed during tests that the `aria-autocomplete` is actually improving the screen reader announce in NVDA ..slightly (no difference in JAWS). It adds an extra 'has autocomplete' to the focus announce.
I've left the attribute in place but made the value dynamic based upon the `autoselect` configuration (and also iOS detection - as the existing util function already considers this).